### DISCUSSION TITLE
Native go strings, GDExtension String and GDExtension StringName

### DISCUSSION THREAD

**Discussion by pcting** [godot-go/godot-go#91] [Polls]
I'm currently thinking of hiding String and StringName in the GDExtensionClasses. I'm looking to get feedback on what everyone thinks.

**Comment by metaleap** [comment]
There's use-cases for both. For rapid-iteration/prototyping, the Go-native `string`s are clearly a most convenient affordance, on the one hand. Yet on the other hand, occasionally and especially in vaster code-bases, some like to keep "boundary"/"outside" things clearly distinct and explicitly recognizable, as that indicates the very foreign-ness this code-path is touching. 

Examples:

- I like my ORM-of-choice's own DBString, DBInt64 etc. types because then one's data structs can have clearly-DB-fields as well as other (runtime-utilized) fields, without needing cumbersome annotations, have-to-maintain field tags, registration code etc.
- it's very fine when generated code has much uglyness and non-idiomatic naming: another immediate recognition that this isn't own-codebase-owned code.

=)
